code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+                        by = c(("Healthboard" = "Healthboard"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+#replace obds in correct column for all reason groups
+datafile27<-datafile27 %>% mutate(OBDs=
+                                    if_else(is.na(OBDs),0L,OBDs))
+
+#remove surplus columns
+datafile27<-select(datafile27,-obds2)
+
+datafile27<-arrange(datafile27,datafile27$Healthboard,datafile27$age_grp,datafile27$reas1)
+
+
+### 17. Save out bed days HB file ----
+
+write.xlsx(datafile27,paste0(filepath,"HB bed days data sheet_R.xlsx"))
+
+#########################################
+### 18. LA bed days  ----
+#########################################
+labedstemplate<-read.csv(paste0(filepath2,"la_template.csv"))
+
+labedstemplate <- labedstemplate %>% 
+  rename(LocalAuthorityArea = LA,
+         age_grp = age)
+
+labedstemplate<-labedstemplate%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+datafile20<-datafile20%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+##################################################################
+
+#amend variables as necessary in order for matchi_numberng to work
+
+labeddaysdatasheetminusstandard <- left_join(datafile20,labedstemplate,
+                                             by = c(("LocalAuthorityArea" = "LocalAuthorityArea"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+arrange(labeddaysdatasheetminusstandard,LocalAuthorityArea,age_grp,reas1) # arrange data in order for tables
+
+#write_sav(labeddaysdatasheetminusstandard,paste0(filepath,"la bed days data sheet minus standard.sav")) # save out file
+labeddaysdatasheetminusstandard<-labeddaysdatasheetminusstandard%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+#Add in total for Standard filter on any row that isn't all
+datafile32a<-filter(labeddaysdatasheetminusstandard,reas1!="ALL") 
+#Rename reason_grp_high_level as Standard where there isn't a Code 9
+datafile32<-datafile32a%>% mutate(reas1=
+                                    if_else(reas1%in%c("HEALTH AND SOCIAL CARE REASONS","PATIENT/CARER/FAMILY-RELATED REASONS"),"STANDARD","CODE 9"))
+table(datafile32$reas1)
+
+#select standard only.
+datafile33<-filter(datafile32,reas1=="STANDARD")
+# aggregate LA Standard delays
+
+lastandard<- datafile33 %>% 
+  group_by(LocalAuthorityArea,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+#datafile24<-datafile24%>%rename(OBDs2=obds_in_month)
+
+lastandardandothers<- bind_rows(lastandard, labeddaysdatasheetminusstandard)
+lastandardandothers<-lastandardandothers%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+
+arrange(lastandardandothers,LocalAuthorityArea,age_grp,reas1) # issue here is that the rows with zeros don't appear so have to match to output file
```

Comment:
See "Style Guide" and "Pipes" sections in general feedback.
Why are we changing these variables to be upper case?
What do you mean "rows with zeros don't appear"?
```suggestion
lastandardandothers <- bind_rows(lastandard, labeddaysdatasheetminusstandard) %>%
    mutate(LocalAuthorityArea = toupper(LocalAuthorityArea), age_grp = toupper(age_grp), reas1 = toupper(reas1)) %>%
    arrange(LocalAuthorityArea, age_grp, reas1) # issue here is that the rows with zeros don't appear so have to match to output file
```